domain/application/service/application.go

```diff
@@ -610,6 +629,23 @@ func (s *ApplicationService) DestroyApplication(ctx context.Context, appName str
 	return errors.Annotatef(err, "destroying application %q", appName)
 }
 
+// EnsureApplicationDead is called by the cleanup worker if a mongo
```

Comment:
`mongo`?

Reply:
Yes, this is used by the mongo cleanup logic. It's needed until all that can be ported across.